docs/HeliumParamSpecs.md

```diff
@@ -0,0 +1,161 @@
+    * Invalid input - empty, foo, values not in valid range [<0.0 , >10.0]
+         * Status : 400
+         * Content-Type: text/plain
+         * Response Body: Invalid rating parameter
+
+
+* Name : actorid
+* Type: string
+* Parameter Validations
+    * Valid input : Wellformed actorid rule - starts with 'nm' followed by upto a 9 digit integer converted to string. Example 'nm9877392'
+         * Status : 200
+         * Content-Type: application/json
+         * Response Body: List of movies filtered by actorid as [{}, {},..] , empty [] when no results
+    * Invalid input - empty, foo, values not conforming to wellformed actorid rule.  Example: 'nm1234', 'ab1234'
+         * Status : 400
+         * Content-Type: text/plain
+         * Response Body: Invalid actorid parameter
+
+
+* Name : genre
+* Type: string
+* Parameter Validations:
+    * Valid input - genre length >= 3. Example 'War' , 'Horror'
+         * Status : 200
+         * Content-Type: application/json
+         * Response Body: List of movies filtered by actorid as [{}, {},..] , empty [] when no results
+    * Invalid input - empty, genre length < 3 characters
+         * Status : 400
+         * Content-Type: text/plain
+         * Response Body: Invalid genre parameter
+
+
+* Name : pagesize
+* Type: integer
+* Parameter Validation:
+    * Default value: 100
+    * Valid input range : [1, 1000]
+         * Status : 200
+         * Content-Type: application/json
+         * Response Body: List of movies filtered by pagesize as [{}, {},..], empty [] when no results
+    * Invalid input - Non integers are invalid. Example : 100.23
+         * Status : 400
+         * Content-Type: text/plain
+         * Response Body: Invalid pagesize parameter
+
+
+* Name : pagenumber
+* Type: integer
+* Parameter Validations:
+    * Default value: 1
+    * Valid input range : [1, 10000]
+         * Status : 200
+         * Content-Type: application/json
+         * Response Body: List of movies filtered by pagenumber as [{}, {},..], empty [] when no results
+    * Invalid input - Non integers are invalid. Example : 100.23
+         * Status : 400
+         * Content-Type: text/plain
+         * Response Body: Invalid pagenumber parameter
+
+
+## Multiple Error Scenarios
+
+* Add each error to the errors array in alphabetical order.
+
+### API Parameters
+
+* API /api/movies/{movieid}
+* Name : movieid
+* Type: string
+* Parameter Validations:
+    * Valid input - Wellformed movieid rule - starts with 'tt' followed by upto a 9 digit integer converted to string. Example 'tt0114746'
+         * Status : 200
+         * Content-Type: application/json
+         * Response Body: Single movie document as JSON
+    * Invalid input - empty, movieid does not exist, movieid not conforming to wellformed movieid rule example - 'tt1234', 'nm1234'
+         * Status : 404
+         * Content-Type: text/plain
+         * Response Body: movieid not found
+
+### Common Status code rules followed :
+
```

Comment:
I think the headings here are a little off when I look at it rendered, try : 
## Common Status code rules...
...
### Searches
...
### Single Reads